src/sql/workbench/contrib/commandLine/electron-browser/commandLine.ts

```diff
@@ -31,17 +31,61 @@ import { IEnvironmentService, INativeEnvironmentService } from 'vs/platform/envi
 import { NativeParsedArgs } from 'vs/platform/environment/common/argv';
 
 export interface SqlArgs {
+	/**
+	 * Used to determine file paths to be opened with SQL Editor.
+	 * If provided, we connect the given profile to to it.
+	 * More than one files can be passed to connect to provided profile.
+	 */
 	_?: string[];
+	/**
+	 * Provide authenticationType to be used.
+	 * accepted values: AzureMFA, SqlLogin, Integrated, etc.
+	 */
 	authenticationType?: string
+	/**
+	 * Name of database
+	 */
 	database?: string;
+	/**
+	 * Name of server
+	 */
 	server?: string;
+	/**
+	 * User name/email address
+	 */
 	user?: string;
+	/**
+	 * Operation to perform:
+	 * accepted values: connect, openConnectionDialog
+	 */
 	command?: string;
+	/**
+	 * Name of connection provider,
+	 * accepted values: mssql (by default), pgsql, etc.
+	 */
 	provider?: string;
-	aad?: boolean; // deprecated - used by SSMS - authenticationType should be used instead
-	integrated?: boolean; // deprecated - used by SSMS - authenticationType should be used instead.
+	/**
+	 * Deprecated - used by SSMS - authenticationType should be used instead
+	 */
+	aad?: boolean;
+	/**
+	 * Deprecated - used by SSMS - authenticationType should be used instead.
+	 */
+	integrated?: boolean;
+	/**
+	 * Whether or not to show dashboard
+	 * accepted values: true, false (by default).
+	 */
 	showDashboard?: boolean;
+	/**
+	 * Supports providing applicationName that will be used for connection profile app name.
+	 */
 	applicationName?: string;
+	/**
+	 *  Supports providing advanced connection properties that providers support.
+	 *  Value must be a json object containing key-value pairs in format: '{"key1":"value1","key2":"value2",...}'
+	 */
+	properties?: string;
```

Comment:
These options aren't for URIs only - they are valid for the command line as well. 

e.g. `azuredatastudio --server localhost --provider mssql --user sa`

This is why I suggested `connectionOptions` (although if you wanted to use `connectionProperties` as well that'd be fine)

The other things are a bit more clearly SQL connection specific, but just `properties` is so generic it would be best to make it very explicit IMO.

Reply:
Ok, makes sense. Will use connectionProperties then.